src/arithmetic/algebraic_expression.h

```diff
@@ -4,108 +4,266 @@
 * This file is available under the Redis Labs Source Available License Agreement
 */
 
-#ifndef ALGEBRAIC_EXPRESSION_H
-#define ALGEBRAIC_EXPRESSION_H
+#pragma once
 
 #include "../graph/query_graph.h"
 #include "../graph/graph.h"
 
+static GrB_Matrix IDENTITY_MATRIX = (GrB_Matrix)0x31032017;  // Identity matrix.
+
 // Matrix, vector operations.
 typedef enum {
-	AL_EXP_UNARY,
-	AL_EXP_ADD,
-	AL_EXP_MUL,
-	AL_EXP_TRANSPOSE,
+	AL_EXP_ADD = 1,                 // Matrix addition.
+	AL_EXP_MUL = (1 << 1),          // Matrix multiplication.
+	AL_EXP_POW = (1 << 2),          // Matrix raised to a power.
+	AL_EXP_TRANSPOSE = (1 << 3),    // Matrix transpose.
 } AL_EXP_OP;
 
+#define AL_EXP_ALL (AL_EXP_ADD | AL_EXP_MUL | AL_EXP_POW | AL_EXP_TRANSPOSE)
+
 // Type of node within an algebraic expression
 typedef enum {
 	AL_OPERAND,
 	AL_OPERATION,
-} AlgebraicExpressionNodeType;
+} AlgebraicExpressionType;
 
 /* Forward declarations. */
-typedef struct AlgebraicExpressionNode AlgebraicExpressionNode;
+typedef struct AlgebraicExpression AlgebraicExpression;
 
-struct AlgebraicExpressionNode {
+struct AlgebraicExpression {
 	union {
-		GrB_Matrix operand;
+        struct {
+            bool free;              // Should the matrix be freed?
```

Comment:
Reminder that we discussed removing this!

Reply:
👍